feature/floor-map/src/main/java/io/github/droidkaigi/confsched2023/floormap/FloorMapScreen.kt

```diff
@@ -172,42 +194,65 @@ private fun FloorMapContent(
             )
         } else {
             MobileContent(
+                nestedScrollConnection = nestedScrollConnection,
                 uiState = uiState,
                 onSideEventClick = onSideEventClick,
-            )
-            FloorLevelSwitcher(
-                selectingFloorLevel = uiState.floorLevel,
                 onClickFloorLevelSwitcher = onClickFloorLevelSwitcher,
-                modifier = Modifier
-                    .align(Alignment.BottomCenter)
-                    .padding(bottom = 24.dp),
             )
         }
     }
 }
 
 @Composable
 private fun MobileContent(
+    nestedScrollConnection: NestedScrollConnection,
     uiState: FloorMapScreenUiState,
     onSideEventClick: (url: String) -> Unit,
+    onClickFloorLevelSwitcher: (FloorLevel) -> Unit,
 ) {
-    Column(
-        modifier = Modifier.fillMaxSize(),
+    val density = LocalDensity.current
+    var boxHeight by remember { mutableIntStateOf(0) }
+    var switcherPosition by remember { mutableFloatStateOf(0f) }
+    val gradientStartRatio by remember {
+        derivedStateOf {
+            if (boxHeight == 0 || switcherPosition == 0f) {
+                1f
+            } else {
+                switcherPosition / (boxHeight - with(density) { 24.dp.toPx() })
+            }
+        }
+    }
+    Box(
+        Modifier
+            .onGloballyPositioned {
+                boxHeight = it.size.height
+            },
     ) {
-        FloorMap(uiState = uiState.smallFloorMapContentUiState.floorMapUiState)
-        FloorMapSideEventList(
-            uiState = uiState.smallFloorMapContentUiState.sideEventListUiState,
+        FloorMapAndSideEventList(
+            floorMapUiState = uiState.smallFloorMapContentUiState.floorMapUiState,
+            sideEventListUiState = uiState.smallFloorMapContentUiState.sideEventListUiState,
             onSideEventClick = onSideEventClick,
             modifier = Modifier
-                .weight(1f)
+                .fillMaxSize()
                 .padding(bottom = 56.dp)
+                .nestedScroll(nestedScrollConnection)
                 .fadingEdge(
                     Brush.verticalGradient(
-                        0.85f to Color.Black,
+                        gradientStartRatio to Color.Black,
```

Comment:
To set the edge gradient appropriately regardless of screen size or orientation, the gradient start position is calculated and set based on the floor switcher and screen size.

Reply:
I didn't notice this! Looks great!